pkg/minikube/cluster/cluster.go

```diff
@@ -308,6 +308,8 @@ func (m *MachineConfig) CacheMinikubeISOFromURL() error {
 	response, err := http.Get(m.MinikubeISO)
 	if err != nil {
 		return err
+	} else if response.StatusCode != http.StatusOK {
```

Comment:
This whole conditional block could be reformatted to reduce the indentations as something like (also defer closing response body should be done ASAP after checking GET error):

```go
response, err := http.Get(m.MinikubeISO)
if err != nil {
	return err
}

defer response.Body.Close()

if response.StatusCode != http.StatusOK {
	return fmt.Errorf("Received %d response from %s while trying to download minikube.iso", response.StatusCode, m.MinikubeISO)
}

out, err := os.Create(m.GetISOCacheFilepath())
if err != nil {
	return err
}
defer out.Close()
if _, err = io.Copy(out, response.Body); err != nil {
	return err
}

return nil